### PR TITLE
Array serialization

### PR DESCRIPTION
**Pull request by shamrin**
This implements Array serialization. Including nested Array-of-References and positional Arrays.

### CODE REVIEW

File: typesystem/fields.py [hunk 1]
```diff
@@ -658,6 +658,19 @@ def validate(self, value: typing.Any, *, strict: bool = False) -> typing.Any:
...
+        if obj is None:
+            return None
+
+        if isinstance(self.items, list):
+            return [
+                serializer.serialize(value)
+                for serializer, value in zip(self.items, obj)
+            ]
+
+        assert self.items is not None
```
Comment (tomchristie) [inline typesystem/fields.py]: This assertion looks odd to me. Looks like `self.items` *can* legitimately be `None` here, no?

Comment (shamrin) [reply typesystem/fields.py]: You are right, thank you! I've fixed it.

**Comment by shamrin** [comment]
@tomchristie Could you look at this again please? I've fixed the PR issue you have raised.

**Comment by tomchristie** [comment]
Thanks!

**Pull request merged**